src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,295 @@
+            {
+                // If there are unprocessed chars, encode and write them as the final block.
+                ReadOnlyMemory<char> charsNotProcessedFromPreviousChunk = this.buffer;
+
+                if (!charsNotProcessedFromPreviousChunk.IsEmpty)
+                {
+                    int firstIndexToEscape = this.jsonWriter.NeedsEscaping(charsNotProcessedFromPreviousChunk.Span);
+                    WriteChunk(charsNotProcessedFromPreviousChunk.Span, isFinalBlock: true, firstIndexToEscape);
+
+                    // Clear the buffer since all bytes have been written.
+                    buffer = Array.Empty<char>();
+                }
+
+                await this.jsonWriter.FlushAsync();
+            }
+
+            /// <summary>
+            /// Writes a specified number of characters from the given character array to the ODataUtf8JsonWriter.
+            /// </summary>
+            /// <param name="value">The character array from which to write characters.</param>
+            /// <param name="index">The starting index in the character array from which to begin writing characters.</param>
+            /// <param name="count">The number of characters to write from the character array.</param>
+            public override void Write(char[] value, int index, int count)
+            {
+                ReadOnlySpan<char> charValuesToWrite = value.AsSpan().Slice(index, count);
+                this.StartWritingCharValuesInChunks(charValuesToWrite);
+            }
+
+            /// <summary>
+            /// Asynchronously writes a specified number of characters from the given character array to theODataUtf8JsonWriter.
+            /// </summary>
+            /// <param name="value">The character array from which to write characters.</param>
+            /// <param name="index">The starting index in the character array from which to begin writing characters.</param>
+            /// <param name="count">The number of characters to write from the character array.</param>
+            /// <returns>A task representing the asynchronous write operation.</returns>
+            public override async Task WriteAsync(char[] value, int index, int count)
+            {
+                ReadOnlyMemory<char> charValuesToWrite = buffer.AsMemory().Slice(index, count);
+                await this.StartWritingCharValuesInChunksAsync(charValuesToWrite);
+            }
+
+            /// <summary>
+            /// Writes the characters represented by the provided read-only span in chunks, processing them for escaping if necessary, and flushing the writer if the buffer threshold is reached.
+            /// </summary>
+            /// <param name="value">The read-only span containing the characters to be written.</param>
+            private void StartWritingCharValuesInChunks(ReadOnlySpan<char> value)
+            {
+                // Process the input value in chunks
+                for (int i = 0; i < value.Length; i += chunkSize)

```

Comment:
Where does `chunkSize` come from?

Reply:
It is in the `ODataUtf8JsonWriter` class.

Reply:
private const int chunkSize = 2048;